pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -228,18 +236,24 @@ func (rq *ResourceQuotaController) worker(queue workqueue.RateLimitingInterface)
 // Run begins quota controller using the specified number of workers
 func (rq *ResourceQuotaController) Run(workers int, stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
-	go rq.rqController.Run(stopCh)
+
+	glog.Infof("Starting resource quota controller")
+
 	// the controllers that replenish other resources to respond rapidly to state changes
 	for _, replenishmentController := range rq.replenishmentControllers {
 		go replenishmentController.Run(stopCh)
 	}
+
+	if !cache.WaitForCacheSync(stopCh, rq.informerSyncedFuncs...) {
```

Comment:
if it will be a common pattern for every other controller, we can do it in a follow-on and possible put it behind a shared library call.  my understanding is that the wait is infinite, unless it gets something on the stopCh which is a shutdown of controller anyway.